### PR TITLE
move quarto render into cleanup script

### PR DESCRIPTION
**Pull request by pawelru**
fix https://github.com/insightsengineering/tlg-catalog/issues/272

move `quarto render` into R CMD BUILD:
```
❯ R CMD BUILD .
* checking for file ‘./DESCRIPTION’ ... OK
* preparing ‘tlg.catalog.pkg’:
* checking DESCRIPTION meta-information ... OK
* running ‘cleanup’                                                            <<<<------ HERE!
* checking for LF line-endings in source and make files and shell scripts
* checking for empty or unneeded directories
* building ‘tlg.catalog.pkg_0.1.0.9029.tar.gz’
```

After `quarto render` we remove the symlink to the book.
As a result, we have refreshed files for snapshot tests and also we don't have a book files that R CMD CHECK complains about.

### CODE REVIEW

**Comment by pawelru** [comment]
This looks good and it's ready for a review.

Unfortunately the logs on the `main` branch are unavailable. I have created a dummy PR to trigger CI so as to be able to compare the output - in particular the scope of the tests. These should be identical. Please open the logs then "Show R CMD CHECK logs" then "tlg.catalog.pkg.Rcheck/tests/testthat.Rout"

- here:
  - R CMD CHECK devel
  https://github.com/insightsengineering/tlg-catalog/actions/runs/11348872739/job/31563675696
    ```
    [ FAIL 0 | WARN 0 | SKIP 88 | PASS 252 ]
    ```
  - R CMD CHECK stable
  https://github.com/insightsengineering/tlg-catalog/actions/runs/11348872739/job/31563676598
    ```
     [ FAIL 0 | WARN 0 | SKIP 88 | PASS 252 ]
    ```

- branch `test_no_changes`
  - R CMD CHECK devel
  https://github.com/insightsengineering/tlg-catalog/actions/runs/11350254923/job/31568241763
    ```
    [ FAIL 0 | WARN 0 | SKIP 88 | PASS 252 ]
    ```
  - R CMD CHECK stable
  https://github.com/insightsengineering/tlg-catalog/actions/runs/11350254923/job/31568242157
    ```
     [ FAIL 0 | WARN 0 | SKIP 88 | PASS 252 ]
    ```

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -32,6 +32,7 @@ jobs:
       additional-env-vars: |
         NOT_CRAN=true
         QUARTO_PROFILE=development
+        TLG_CATALOG_PKG_BUILD_RENDER=TRUE
```
Comment (pawelru) [inline .github/workflows/check.yml:35]: @cicdguy 
if we are going with this approach - we would have to add that flag to our integration tests

**Comment by m7pr** [comment]
I can confirm there is no Quarto rendering on the first build
image

But I can see Quarto rendering on the other one

image

**Comment by m7pr** [comment]
I have read the CLA Document and I hereby sign the CLA

File: .github/workflows/check.yml (hunk 1 above)
Comment (pawelru) [reply .github/workflows/check.yml:35]: Hey @cicdguy. Ping on this. I got this merged and we need updates in our integration test pipelines.

**Pull request merged**